experiment/kind-detect-local-e2e.sh

```diff
@@ -0,0 +1,289 @@
+#!/bin/sh
```

Comment:
This is exactly the same as https://github.com/kubernetes-sigs/kind/blob/91cc28782d7343dc1547c3d40e7136fd24ed5ddc/hack/ci/e2e-k8s.sh except for the changes noted below.